pytorch_pfn_extras/onnx/_grad.py

```diff
@@ -0,0 +1,98 @@
+from typing import Generator, List, Optional, Tuple
+from contextlib import contextmanager
+
+import torch
+import torch.onnx
+import threading
+from pytorch_pfn_extras.onnx._as_output import as_output
+
+
+_grad_state = threading.local()
+
+
+@contextmanager
+def init_grad_state() -> Generator[None, None, None]:
+    _grad_state.n_grad_call = 0
+    try:
+        yield
+    finally:
+        _grad_state.n_grad_call = None
+
+
+def grad(
+    output: torch.Tensor,
+    inputs: Tuple[torch.Tensor, ...],
+    retain_graph: Optional[bool] = None,
+    create_graph: bool = False,
+    allow_unused: bool = False,
+    only_inputs: bool = True,
+) -> Tuple[Optional[torch.Tensor], ...]:
+    grad_output = torch.ones_like(output)
+
+    if torch.jit.is_tracing():
+        assert hasattr(_grad_state, "n_grad_call")
+        n_grad_call = _grad_state.n_grad_call
+        _grad_state.n_grad_call += 1
+
+        output_name = f"Gradient_y_{n_grad_call}"
+        output = as_output(output_name, output)
+
+        assert only_inputs, "only_inputs=False case is not supported now"
+
+        input_names = []
+        inputs_l = list(inputs)
+        for i, input in enumerate(inputs):
+            input_name = f"Gradient_x_{i}_{n_grad_call}"
+            input_names.append(input_name)
+            inputs_l[i] = as_output(input_name, input)
+
+        class _Gradient(torch.autograd.Function):
+            @staticmethod
+            def forward(
+                ctx,
+                output: torch.Tensor,
+                grad_output: Optional[torch.Tensor],
+                *inputs: Tuple[torch.Tensor, ...],
+            ):
+                @torch.jit.script
+                def _grad(
+                    output: torch.Tensor,
+                    inputs: List[torch.Tensor],
+                    grad_output: Optional[torch.Tensor],
+                    retain_graph: Optional[bool],
+                    create_graph: bool,
+                    allow_unused: bool,
+                ):
+                    return torch.autograd.grad(
+                        outputs=[output],
+                        inputs=inputs,
+                        grad_outputs=[grad_output],
```

Comment:
This should work? (align with the code in non-tracing case)

```suggestion
                        grad_outputs=grad_output,
```

Reply:
no, torch script supports only the list case.

Reply:
I see, thanks for clarificaiton!